articles/css-in-js-solutions/index.md

```diff
@@ -0,0 +1,121 @@
+
+This gap in the simplification of your codebase concerning your CSS files is why you need to seek out means of encapsulating your CSS code within your JavaScript files, rather than creating external stylesheets for each web component.
+
+Although web components simplify the development process of web applications, with it comes the barrier of maintaining multiple external style sheets when using CSS. Nonetheless, some solutions exist to simplify CSS stylings when using web components, such as CSS-in-JS.
+
+### What is CSS-in-JS, and how is it salutary to CSS for Web Components? 
+
+CSS-in-JS is a JavaScript library that proffers an ample variety of styling techniques in styling web components. CSS-in-JS was a result of the flaws in using CSS in building web applications, especially with custom elements, and some of which include scoping, specificity, portability, and dynamic functioning.
+
+The adaptation of CSS-in-JS libraries over CSS is as a result of stylesheets loading in any order, which sometimes override each other. Meanwhile, before the advent of CSS-in-JS, developers applied specific CSS methodologies to ensure modular, reusable, and scalable styling, and some of the methodologies are BEM, OOCSS, and SMACSS amongst others.
+
+CSS-in-JS is salutary over CSS for various reasons, but the end goal is a result of the need for a simpler, easier to maintain codebase. Rather than maintaining multiple external stylesheets, CSS-in-JS solutions provide us with different libraries with specific advantages for developers to choose from in the development of their web applications.
+
+CSS-in-JS is the best styling technique to settle for in the development of web applications with web components. But, this doesn’t render CSS redundant because it certainly does have its pros and its cons.
+
+![A code example of style declaration in the JavaScript variable font](/engineering-education/css-in-js-solutions/js.png) 
+
+## Pros of CSS
+
+1. Inline Styling: The use of inline styles enhances website performance because the browser loads the CSS within elements faster before loading external styles.
+
+2. Reduction in Redundancy: With inline styles, you can avoid redundant code and only declare CSS styles to a particular element when needed, this way, all extraneous code stays out of the code and at the sympathy of the delete key.
+3.  External stylesheet: Inline-styles aren't the only golden property of CSS.  External stylesheet keeps your codebase organized and allows you to store all CSS code within a single file and reference them in documents that require them.
```

Comment:
Inline-styles aren't the only **way to write CSS.**

External stylesheet**s**

reference them in **webpages** that need them.

Reply:
External stylesheets **keep** your codebase

Remove s in keeps